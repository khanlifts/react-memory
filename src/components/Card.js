import React from 'react';

export default class Card extends React.Component {
  state = {
    flipActive: false,
    deleteButtonActive: false
  };
  handleFlip = () => {
    const currentState = this.state.flipActive
    this.setState(() => ({ flipActive: !currentState }));
    this.props.handleSameContent(this.props.backContent);
  };
  handleDisappear = (e) => {
    e.stopPropagation();
    const deleteBtnState = this.state.deleteButtonActive;
    this.setState(() => ({ deleteButtonActive: !deleteBtnState }));
  };
  render() {
    return (
      <div>
        <div className={this.state.deleteButtonActive ? 'container-hide' : null}>
          <section className="container">
            <div
              id="card"
              className={this.state.flipActive ? 'flipped' : null}
              onClick={this.handleFlip}>
              <figure className="front"><span>{this.props.count}</span></figure>
              <figure className="back">
                <span>{this.props.backContent}</span>
                <button
                  className="card__button"
                  onClick={(e) => {
                    e.stopPropagation();
                    this.props.handleDeleteContent(this.props.backContent);
                  }}
                  >x
                </button>
              </figure>
            </div>
          </section>
        </div>
      </div>
    );
  }
};
