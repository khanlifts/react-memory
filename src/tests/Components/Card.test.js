import React from 'react';
import { shallow } from 'enzyme';
import Card from '../../components/Card';

test('should render Grid correctly', () => {
  const wrapper = shallow(<Card />);
  // expect(wrapper.find('button').text()).toBe('x');
  expect(wrapper).toMatchSnapshot();
  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Card />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
  //
  // console.log(renderer.getRenderOutput());
});
