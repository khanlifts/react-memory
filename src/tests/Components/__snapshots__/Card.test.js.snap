// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Grid correctly 1`] = `
<div>
  <div
    className={null}
  >
    <section
      className="container"
    >
      <div
        className={null}
        id="card"
        onClick={[Function]}
      >
        <figure
          className="front"
        >
          <span />
        </figure>
        <figure
          className="back"
        >
          <span />
          <button
            className="card__button"
            onClick={[Function]}
          >
            x
          </button>
        </figure>
      </div>
    </section>
  </div>
</div>
`;
